pkg/cmd/container/cp_linux.go

```diff
@@ -39,17 +40,22 @@ func Cp(ctx context.Context, client *containerd.Client, options types.ContainerC
 				ctx,
 				client,
 				found.Container,
-				options.Container2Host,
-				options.DestPath,
-				options.SrcPath,
-				options.GOptions.Snapshotter,
-				options.FollowSymLink)
+				options)
 		},
 	}
 	count, err := walker.Walk(ctx, options.ContainerReq)
 
-	if count < 1 {
-		err = fmt.Errorf("could not find container: %s, with error: %w", options.ContainerReq, err)
+	if count == -1 {
```

Comment:
I don't quite follow this change. What's the issue with the current logic? (also I don't think we should panic here).

Reply:
Current logic (count < 1) would treat identically two unrelated conditions:
- count == 0, which means we found no container
- count == -1, which means we errored 

https://github.com/containerd/nerdctl/blob/main/pkg/idutil/containerwalker/containerwalker.go#L47-L74